src/nvim/event/process.c

```diff
@@ -221,29 +221,15 @@ void process_stop(Process *proc) FUNC_ATTR_NONNULL_ALL
   }
   proc->stopped_time = os_hrtime();
 
-  switch (proc->type) {
-    case kProcessTypeUv:
-      // Close the process's stdin. If the process doesn't close its own
-      // stdout/stderr, they will be closed when it exits(possibly due to being
-      // terminated after a timeout)
-      stream_may_close(&proc->in);
-      os_proc_tree_kill(proc->pid, SIGTERM);
-      break;
-    case kProcessTypePty:
-      // close all streams for pty processes to send SIGHUP to the process
-      process_close_streams(proc);
-      pty_process_close_master((PtyProcess *)proc);
-      break;
-    default:
-      abort();
-  }
+  assert(proc->type == kProcessTypeUv || proc->type == kProcessTypePty);
+  os_proc_tree_kill(proc->pid, SIGTERM);
```

Comment:
I think some people expect SIGHUP as the primary signal when a pty/terminal is closed. Why change that? https://github.com/neovim/neovim/issues/5619

Reply:
I've thought it might be more consistent - initially I've noticed it in a Neomake test that could not trap SIGTERM anymore when using `pty=1` IIRC (which confused me).
Thanks for the references, will follow up / revert this then eventually.

Reply:
@hardenedapple @bfredl 
What do you think about sending SIGTERM first, and only SIGHUP after 2s (when the job is still there)?  (via https://github.com/neovim/neovim/issues/5619)

/cc @jamessan via https://github.com/neovim/neovim/issues/8760#issuecomment-406380765
As seen in https://github.com/neovim/neovim/issues/8760#issuecomment-406364645 sending SIGTERM additionally makes sense (I assume that wasn't the case back then maybe?).  This is about the order it should happen in.

I see SIGHUP here similar to closing stdin, which might cause a job to exit already, and think that using SIGTERM first is more explicity.  For non-pty jobs it makes the outcome more predictable (when closing stdin and sending SIGTERM is done together).

Reply:
> and only SIGHUP after 2s

Isn't the point to follow expected conventions? I don't know how important it is, but I would guess sending SIGTERM first will yield the same bug reports as before.

Reply:
I've thought there wasn't any SIGHUP before?
(Also https://github.com/neovim/neovim/issues/8760#issuecomment-406364645 actually did not expect/handle SIGHUP.)

> Isn't the point to follow expected conventions?

Sure.  But I am not sure what those are.

Reply:
#8760 is of course the opposite request, but... Some people are also surprised when shell commands return 0 for success, until they learn they conventions.

SIGHUP convention is more obscure, but all terminals are consistent about it AFAIK.

Reply:
#8760 is of course the opposite request, but... Some people are also surprised when shell commands return 0 for success, until they learn they conventions.

SIGHUP convention is more obscure, but all terminals are consistent about it AFAIK.

Reply:
What is obscure about SIGHUP? It
is defined specifically for the situation of closed terminal. IIRC it is even sent by kernel itself when master PTY is closed. Behaving differently than most other pty hosts by default is what I would call "obscure".

Reply:
The point of using a pty job is to get pty semantics. If you do not want those, don't use a pty job or override the semantics manually.

Reply:
I am not saying to not send SIGHUP - but only after / when SIGTERM is not handled.

Reply:
> The point of using a pty job is to get pty semantics.

Not necessarily - I thought about using it only for merging stdout/stderr in the first place.
I get your point though.  I will back it out of here.

Reply:
> Not necessarily - I thought about using it only for merging stdout/stderr in the first place

Didnt we already discuss implementing this properly by just sharing a uv pipe? Merging stderr with stdout is an independent desicion from running with a controlling pty. We should enable additional behaviors by adding new options, not by distorting the defaults for non-standard use cases. Most terminals don't hunt down and SIGTERM/KILL their child processes, especially not before closing the pty (= SIGHUP).

Reply:
> Didnt we already discuss implementing this properly by just sharing a uv pipe?

Yes, still wanted to give this as an example (again).

> Most terminals don't hunt down and SIGTERM/KILL their child processes

True.
Should we stop doing this then also?

> , especially not before closing the pty (= SIGHUP).

That wasn't clear to me.

However `process_close_streams(proc);` is still called before closing the master - which is similar to closing stdin for non-pty before SIGTERM, no?  I.e. there are inconsistent outcomes there then also (the process might exit itself due to stdin being closed, instead of because of SIGHUP).

Reply:
> Should we stop doing this then also?

I thought I/we implemented this long ago (perhaps one of the linked issues, but my travel battery is a bit low so I won't read them all through now).

> I.e. there are inconsistent outcomes there then also (the process might exit itself due to stdin being closed, instead of because of SIGHUP).

AFAIK this is up to the client process and not our concern. We just close the master end ( `process_close_streams` are just duplicates of the same master handle to make the code simpler IIRC), and the process might choose to handle SIGHUP or be killed by it, or silence it and exit by stdin read failure, or just ignore it altogheter etc.

Reply:
The master code (and the latest version of this PR, which doesn't change pty code anymore) looks correct to me.